x-pack/plugins/security_solution/public/detections/components/alerts_table/timeline_actions/use_investigate_in_timeline.tsx

```diff
@@ -52,9 +52,27 @@ export const useInvestigateInTimeline = ({
 
   const getExceptionFilter = useCallback(
     async (ecsData: Ecs): Promise<Filter | undefined> => {
-      const exceptionsLists = (getField(ecsData, ALERT_RULE_EXCEPTIONS_LIST) ?? []).reduce(
-        (acc: ExceptionListId[], next: string) => {
-          const parsedList = JSON.parse(next);
+      // This pulls exceptions list information from `_source`
+      // This primarily matters for the old `signal` alerts a user may be viewing
+      // as new exception lists are pulled from kibana.alert.rule.parameters[0].exception_lists;
+      // Source was removed in favour of the fields api which passes the exceptions_list via `kibana.alert.rule.parameters`
+      let exceptionsList = getField(ecsData, ALERT_RULE_EXCEPTIONS_LIST) ?? [];
+
+      if (exceptionsList.length === 0) {
+        try {
+          const ruleParameters = getField(ecsData, ALERT_RULE_PARAMETERS) ?? {};
+          if (ruleParameters.length > 0) {
+            const parametersObject = JSON.parse(ruleParameters[0]);
+            exceptionsList = parametersObject?.exceptions_list ?? [];
+          }
+        } catch (error) {
+          // do nothing, just fail silently as parametersObject is initialized
+        }
+      }
+      const detectionExceptionsLists = exceptionsList.reduce(
+        (acc: ExceptionListId[], next: string | object) => {
+          // parsed rule.parameters returns an object else use the default string representation
+          const parsedList = typeof next === 'string' ? JSON.parse(next) : next;
```

Comment:
Do you think that this line(`JSON.parse`) might result in exception when ......`next === ''`

Reply:
I'm not sure exceptions list will ever actually be `''`, but good to guard against, will add a change for this as well, thanks!